pkg/chunk/purger/request_handler.go

```diff
@@ -5,22 +5,48 @@ import (
 	"fmt"
 	"net/http"
 
+	"github.com/prometheus/client_golang/prometheus"
+
 	"github.com/cortexproject/cortex/pkg/util"
 
 	"github.com/prometheus/common/model"
 	"github.com/prometheus/prometheus/promql"
 	"github.com/weaveworks/common/user"
 )
 
+type deleteRequestHandlerMetrics struct {
+	deleteRequestsReceivedTotal *prometheus.CounterVec
+}
+
+func newDeleteRequestHandlerMetrics(r prometheus.Registerer) *deleteRequestHandlerMetrics {
+	m := deleteRequestHandlerMetrics{}
+
+	m.deleteRequestsReceivedTotal = prometheus.NewCounterVec(prometheus.CounterOpts{
+		Namespace: "cortex",
+		Name:      "purger_delete_requests_received_total",
```

Comment:
Given this metric is tracked only in case it's successfully enqueued, have you considered calling it `purger_delete_requests_enqueued_total`? I'm asking because "received" makes me thing it counts the total number of received requests (including failing ones) but that's not what it does.

Reply:
Actually it is for counting all the received requests including successful, failed and cancelled(yet to be implemented) ones. We can use it to determine the usage, rate of requests we are getting vs processing etc. Does it make sense?